6.5.group_by.py

```diff
@@ -0,0 +1,27 @@
+# Author: Omer Gertler
+
+
+def group_by(f, iterable):
+    """
+    Implementation of group-by method.
+    Make a dictionary when the keys are the return values of "f" with the
+    parameters of "iterable" items. The values are the iterable-items that
+    belong to the key.
+    Example: input: group_by(len, ["hi", "bye", "yo", "try"])
+             output: {2: ['hi', 'yo'], 3: ['bye', 'try']}
+    :param f: function to group-by it
+    :param iterable: items for grouping
+    :return: dictionary {<group-bi-it value>: [<items>]}
+    """
+    d = {}
+    for i in iterable:
+        key = f(i)
+        if key not in d.keys():
+            d[key] = [i]
+        else:
+            d[key].append(i)
+    return d
```

Comment:
replace 'd' 'i' 'f' with more explanatory names
Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing